src/runtimes/node/finder.js

```diff
@@ -6,19 +6,42 @@ const locatePath = require('locate-path')
 
 const pLstat = promisify(lstat)
 
-const allowedExtensions = new Set(['.cjs', '.js', '.mjs', '.ts', '.zip'])
+// List of extensions that this runtime will look for, in order of precedence.
+const allowedExtensions = ['.js', '.zip', '.cjs', '.mjs', '.ts']
+
+// Sorting function, compatible with the callback of Array.sort, which sorts
+// entries by extension according to their position in `allowedExtensions`.
+// It places extensions with a higher precedence last in the array, so that
+// they "win" when the array is flattened into a Map.
+const sortByExtension = (fA, fB) => {
+  const indexA = allowedExtensions.indexOf(fA.extension)
+  const indexB = allowedExtensions.indexOf(fB.extension)
+
+  return indexB - indexA
+}
 
 const findFunctionsInPaths = async function (paths) {
   const functions = await Promise.all(paths.map(getFunctionAtPath))
 
-  // We sort the functions array to put directories first. This is so that
-  // `{name}/{name}.js` takes precedence over `{name}.js`.
-  //
   // It's fine to mutate the array since its scope is local to this function.
   // eslint-disable-next-line fp/no-mutating-methods
-  const sortedFunctions = functions
-    .filter(Boolean)
-    .sort((fA, fB) => Number(fA.stat.isDirectory()) - Number(fB.stat.isDirectory()))
+  const sortedFunctions = functions.filter(Boolean).sort((fA, fB) => {
+    // We first sort the functions array to put directories first. This is so
+    // that `{name}/{name}.js` takes precedence over `{name}.js`.
+    const directorySort = Number(fA.stat.isDirectory()) - Number(fB.stat.isDirectory())
+
+    if (directorySort) {
```

Comment:
```suggestion
    if (directorySort !== 0) {
```

Should we explicit about the typecasting?

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/374/commits/4e6e530a5e1d417baf4a566e52f89d87899b5c05.

Reply:
(Oh, just realised you made an actual suggestion which I could've committed directly!)